src/genome.py

```diff
@@ -88,154 +97,260 @@ def __str__(self) -> str:
 class ListGenome(Genome):
     """
     Representation of a genome.
-
     Implements the Genome interface using Python's built-in lists
     """
 
     def __init__(self, n: int):
         """Create a new genome with length n."""
-        ...  # FIXME
+        #initialize the genome with no TE's yet.
+        self.nucleotide= ['-'] * n
+        self.TE = {}
+        self.active_TE = []
 
     def insert_te(self, pos: int, length: int) -> int:
         """
         Insert a new transposable element.
-
         Insert a new transposable element at position pos and len
         nucleotide forward.
-
         If the TE collides with an existing TE, i.e. genome[pos]
         already contains TEs, then that TE should be disabled and
         removed from the set of active TEs.
-
         Returns a new ID for the transposable element.
         """
-        ...  # FIXME
-        return -1
+
+        if len(self.TE) != 0:
+            TE_ID = max(self.TE) + 1
+        else:
+            TE_ID = 1
+        self.TE[TE_ID] = length
+        self.active_TE.append(TE_ID)
+        if self.nucleotide[pos] in self.active_TE:
+            self.disable_te(self.nucleotide[pos])
+        self.nucleotide[pos:pos] = [TE_ID]*length
```

Comment:
You start out with `self.nucleotide` holding strings. Now you insert integers. Mixing types like this can often cause trouble. You didn't need strings to begin with, just an integer that indicated that no TE was there. Since your IDs start at 1, zero or a negative number would do the trick.